x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -42,9 +23,30 @@ export function registerCredentialsRoutes({
     },
     enterpriseSearchRequestHandler.createRequest({
       path: '/as/credentials/collection',
-      hasValidData: (body?: ICredentialsResponse) => {
-        return Array.isArray(body?.results) && typeof body?.meta?.page?.total_results === 'number';
-      },
     })
   );
+  router.get(
+    {
+      path: '/api/app_search/credentials/details',
+      validate: false,
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/as/credentials/details',
+    })
+  );
+  router.delete(
+    {
+      path: '/api/app_search/credentials/{name}',
+      validate: {
+        params: schema.object({
+          name: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      return enterpriseSearchRequestHandler.createRequest({
+        path: `/as/credentials/credentials/${request.params.name}`,
```

Comment:
Wait, are you sure this is correct? We don't have `credentials/credentials` twice in our API endpoint do we?

```suggestion
        path: `/as/credentials/${request.params.name}`,
```

Reply:
😲

Reply:
I'm guessing that doesn't even work. Not sure how that got through.

Reply:
Fixed and verified that it works 08ab4f781a2b819a3e195423da4b36587e7add19